control/runner.go

```diff
@@ -301,14 +301,24 @@ func (r *runner) HandleGomitEvent(e gomit.Event) {
 
 		r.availablePlugins.RUnlock()
 		if pool == nil {
+			// if we didn't find the named plugin in one of our pools, then simply return
 			return
 		}
-		// Check for the highest lower version plugin and move subscriptions that
-		// are not bound to a plugin version to this pool.
+		// Otherwise, we need to roll any subscriptions that may have existed for the plugin being
+		// unloaded to a new version (if we can find one). First, check for the highest version of the
+		// plugin being unloaded that is still available (if there is one)
 		plugin, err := r.pluginManager.get(fmt.Sprintf("%s:%s:%d", core.PluginType(v.Type).String(), v.Name, -1))
 		if err != nil {
+			if r.pluginManager.(*pluginManager).isPluginNotFoundErr(err) {
```

Comment:
These are all in the same package so I think you can just do if err == ErrPluginNotFound and get rid of this assertion + the isPluginNotFound function.